wp.js

```diff
@@ -83,6 +83,47 @@ WP.site = function( endpoint ) {
 };
 
 /**
+ * Create a factory method for requests against an arbitrary endpoint base
+ *
+ * @example
+ *
+ *     wp.myPluginResources = wp.endpoint({
+ *       base: 'resources',
+ *       namespace: 'myplugin/v2'
+ *     });
+ *     wp.myPluginResources().get().then( // ...
+ *     wp.myPluginResources().id( 7 ).then( // ...
+ *
+ * @method endpoint
+ * @param {Object} [options] An options hash for a new MediaRequest
+ * @return {MediaRequest} A MediaRequest instance
```

Comment:
Should both of the above lines be updated to `new CollectionRequest` and `A CollectionRequest instance`?